### ISSUE TITLE
Not auto-loading "Unknown (ASS)" subtitles

### ISSUE THREAD

**Issue by im-scared** [ZeroQI/Hama.bundle#386]
Hello, I've been having a consistent problem with anime not auto-loading subtitles. It seems to be any time the subtitle stream comes up as `Unknown (ASS)`. That is when I've noticed it, at least. If the stream has a name like `English` then it does seem to auto-load, but for a lot of my library that's not the case and I have to pause and manually select the sub stream with every episode. It's pretty frustrating.

I couldn't find any info, but I might just not know the right thing to search.

I don't know about every case, but the current anime I ran into this with has MKV files with an embedded SSA sub stream (according to VLC).

**Comment by ZeroQI** [comment]
Hama does not interact with subtitles in any way
SSA/ASS Subtitles has nothing to do with Absolute Series Scanner
Did you try renaming you subtitle xxx ep 15 title.English.ext as per naming convention
Closing as Plex related but not metadata agent related

**Comment by im-scared** [comment]
Oh my apologies. I don't know why I thought Hama was involved in that. Thank you very much for taking the time to help clarify this for me. I'll look up Plex fixes and the sub renaming fix -- I hadn't thought of renaming the subs because that's going to be a lot of work but it may just be what i need to do :c.

Again, thanks for your time.

**Comment by ZeroQI** [comment]
Just create a batch file you put on your desktop and drag and drop subtitles to it and it must treat argument (the filepath) as %1 and rename to filename.English.ext from english.ext
Or search recursively folders and rename sub ext to english.sub.ext if not already done...

**Comment by im-scared** [comment]
The thing I was concerned about was the fact that these are internal subs and I don't have experience with manipulating MKVs and thought it would be something hard to automate. But it turned out to be super easy! I'm really glad you suggested I just fix the names. I'll document my solution for others to reference in the future.

My Plex server is running on a linux environment so I installed the `mkvtoolnix` package which provides `mkvinfo` and more importantly `mkvpropedit`. I then ran the following command:

```
for f in *.mkv; do echo $f; mkvpropedit "$f" --edit track:s1 --set name='English' --set language=eng --edit track:a1 --set name='Japanese' --set language=jpn; done
```

This sets the name and language of the sub track to `English` and the name and language of the audio track to `Japanese` (this series happened to have one audio and one sub track). Note that the language code [must be a particular format](https://gitlab.com/mbunkus/mkvtoolnix/issues/2419) but the name can be anything I think. Also, the `for` loop is probably not needed (could probably do `mkvpropedit *.mkv` but I was paranoid because I haven't used `mkvpropedit` before.
